publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/expandresources/ExpandImportCandidateHandler.java

```diff
@@ -0,0 +1,85 @@
+package no.unit.nva.publication.events.handlers.expandresources;
+
+import static no.unit.nva.publication.events.bodies.ImportCandidateDataEntryUpdate.IMPORT_CANDIDATE_PERSISTENCE;
+import static no.unit.nva.publication.events.handlers.persistence.PersistedDocument.createIndexDocument;
+import static no.unit.nva.publication.events.handlers.persistence.PersistenceConfig.PERSISTED_ENTRIES_BUCKET;
+import static no.unit.nva.s3.S3Driver.GZIP_ENDING;
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.lambda.runtime.Context;
+import java.net.URI;
+import no.unit.nva.events.handlers.DestinationsEventBridgeEventHandler;
+import no.unit.nva.events.models.AwsEventBridgeDetail;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import no.unit.nva.events.models.EventReference;
+import no.unit.nva.expansion.model.ExpandedImportCandidate;
+import no.unit.nva.publication.events.bodies.ImportCandidateDataEntryUpdate;
+import no.unit.nva.publication.events.handlers.persistence.PersistedDocument;
+import no.unit.nva.publication.model.business.ImportCandidate;
+import no.unit.nva.s3.S3Driver;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UnixPath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExpandImportCandidateHandler extends
+                                          DestinationsEventBridgeEventHandler<EventReference, EventReference> {
+
+    private final Logger logger = LoggerFactory.getLogger(ExpandImportCandidateHandler.class);
+    public static final String EVENTS_BUCKET = "EVENTS_BUCKET";
+    public static final String EMPTY_EVENT_TOPIC = "Event.Empty";
+    private final S3Driver s3Reader;
+    private final S3Driver s3Writer;
+
+    @JacocoGenerated
+    protected ExpandImportCandidateHandler() {
+        this(new S3Driver(EVENTS_BUCKET), new S3Driver(PERSISTED_ENTRIES_BUCKET));
+    }
+
+    public ExpandImportCandidateHandler(S3Driver s3Driver, S3Driver s3Writer) {
+        super(EventReference.class);
+        this.s3Reader = s3Driver;
+        this.s3Writer = s3Writer;
+    }
+
+    @Override
+    protected EventReference processInputPayload(EventReference input,
+                                                 AwsEventBridgeEvent<AwsEventBridgeDetail<EventReference>> event,
+                                                 Context context) {
+        var blob = readBlobFromS3(input);
+        var importCandidate = (ImportCandidate) blob.getNewData();
```

Comment:
Did you really need to cast this to ImportCandidate? isn't it already a ImportCandidate? Asking, not criticizing